### PR TITLE
Switch matching client to gRPC

### PR DESCRIPTION
**Pull request by alexshtin**
1. Converted all matching clients to gRPC version.
2. Matching clients talk to gRPC matching server which is still just a proxy to Thrift handler.
3. Thrift handler is not used for any RPC.
4. Moved code related to matching service from `workflowHandler.go` to `workflowHandlerGRPC.go` to remove unnecessary Thrift-proto conversions.
5. Error handling is now tricky because errors also needs to be converted from proto to Thrift and vice versa. This code will gone away when history service is also migrated.
6. `workflowHandler_test.go` now tests `workflowHandlerGRPC.go.`

### CODE REVIEW

File: cmd/server/temporal/server.go [hunk 1]
```diff
@@ -135,7 +135,11 @@ func (s *server) startService() common.Daemon {
 	servicePortMap := make(map[string]int)
 	for roleName, svcCfg := range s.cfg.Services {
 		serviceName := getServiceName(roleName)
-		servicePortMap[serviceName] = svcCfg.RPC.Port
+		if serviceName == common.FrontendServiceName || serviceName == common.MatchingServiceName {
```
Comment (alexshtin) [inline cmd/server/temporal/server.go]: For some reason it worked for frontend even before this change.

File: service/frontend/workflowHandler.go [hunk 2]
```diff
@@ -340,180 +340,15 @@ func (wh *WorkflowHandler) PollForActivityTask(
...
-			wh.GetLogger().Error("PollForActivityTask failed.",
-				tag.WorkflowTaskListName(pollRequest.GetTaskList().GetName()),
-				tag.Value(ctxTimeout),
-				tag.Error(err))
-			return nil, wh.error(err, scope)
-		}
-	}
-
-	return resp, nil
+	panic("not implemented")
```
Comment (alexshtin) [inline service/frontend/workflowHandler.go]: All this code was moved to `workflowHandlerGRPC.go`

File: cmd/server/temporal/server.go (hunk 1 above)
Comment (samarabbas) [reply cmd/server/temporal/server.go]: Probably no one role directly calls frontend

File: common/adapter/error.go [hunk 3]
```diff
@@ -81,3 +83,52 @@ func ToProtoError(in error) error {
 
 	return st.Err()
 }
+
+// ToProtoError converts gRPC error to Thrift error.
+func ToThriftError(st *status.Status) error {
```
Comment (samarabbas) [inline common/adapter/error.go]: I guess this would go away after grpc migration is done?

File: common/util.go [hunk 4]
```diff
@@ -222,6 +233,23 @@ func IsServiceNonRetryableError(err error) bool {
 	return false
 }
 
+// IsServiceNonRetryableError checks if the error is a non retryable error.
+func IsServiceNonRetryableErrorGRPC(err error) bool {
+	if err == context.DeadlineExceeded {
```
Comment (samarabbas) [inline common/util.go]: I'm not sure this is correct.  Deadline exceeded should be retried.

File: common/adapter/error.go (hunk 3 above)
Comment (alexshtin) [reply common/adapter/error.go]: Yes, entire `adapter` will go away.

File: common/util.go (hunk 4 above)
Comment (alexshtin) [reply common/util.go]: Good catch! It seems I messed up while copying all cases from `IsServiceNonRetryableError`.

Comment (alexshtin) [reply common/util.go]: Will fix it in next PR.

**Pull request merged**